### PR TITLE
refactor: flatten dist folders, replace LESS tokens with Custom Properties

### PR DESCRIPTION
**Pull request by ianmcburnie**
- [X] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description

Okay, this is one of the biggest changes to Skin in recent years, as we look to the future of powering not just eBay, but also unbranded/headless component libraries with a fully custom property based token system! Exciting :-)

* Flattens all dist folders, e.g `dist/switch/ds4/switch.css` and `dist/switch/ds6/switch.css` now flattened to a single `dist/switch/switch.css`. **All deltas between design systems are now expressed via custom properties at run time, instead of LESS variables at compile time.**
* Moves all color LESS variables to a new pure CSS properties file under `src/tokens`. Currently organized like so (but subject to change before final release):
   * evo-core.css
   * evo-light.css
   * evo-dark.css
   * ds4-core.css    
* The exisiting `src/less/tokens` folder has been renamed to `src/less/variables`. This renaming accounts for much of the noise in this PR
* Any remaining tokens, such as padding, font-size, etc that have not been converted over to custom props, will live under `src/less/variables` until we gradually move them over to custom props (I doubt we will do it in this release)
* **We can now remove all the utility mixins and their instances and replace them pure CSS var syntax**. This will be done in a follow up PR to reduce noise
* All CSS variable *fallbacks* now removed. This also account for much of the noise in the PR
* We were previously using LESS vars as the CSS custom property fallback, but now they are removed, values must now be explicitly provided with `ebay/skin/evo-tokens/core` or `ebay/skin/ds4-tokens/core` - bringing in the bundle of CSS custom properties. Light and dark mode will operate in a similar manner, e.g `ebay/skin/evo-tokens/light`, `ebay/skin/evo-tokens/dark`. Requesting `ebay/skin/evo-tokens` would bring in all core, light and dark evo tokens (NOTE: not all teams are ready for dark mode, which is why it's important to offer this granularity in our API)
* In v15 we will explore moving all tokens to an external package, e.g. `@ebay/eds` (for eBay design system tokens) 

## Notes

* The tokens files are now pure CSS. This does mean however we lose the ability to use LESS functions such as `shade` and `tint`. We might want to reinstate the source file as LESS and compile to CSS.
* The distributed tokens currently live altogether in a flat folder under `dist/tokens` but this is subject to change before release
* Given that the only difference in the two websites is a single custom property file (`core-ds4.css`), we can potentially remove the DS4 url and have the tokens files dynamically loaded/removed with javascript from a single page.
* Need to check our public LESS API is still present (although there are going to be much fewer exposed in this release - perhaps only mixins)
* Removed the DS4 storybook (the differences between DS6 and DS4 are so small now - just the colours) - but can revisit if we need this.
* Need to check `legacy-theme`, `legacy-textbox` and `square-corner-theme` are still working as expected in conjunction with these changes

## Checklist

- [X] I verify the build is in a non-broken state
- [X] I verify all changes are within scope of the linked issue

- [X] I regenerated all CSS files under dist folder
- [X] I tested the UI in all supported browsers
- [X] I tested the UI in dark mode and RTL mode
- [X] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

File: src/tokens/ds4-core.css [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    --color-blue-1: opacity(var(--color-blue-4), 20%);
+    --color-blue-2: opacity(var(--color-blue-4), 40%);
+    --color-blue-3: opacity(var(--color-blue-4), 60%);
+    --color-blue-4: #0654ba;
+    --color-blue-5: #0654ba;
+    --color-blue-6: #00489f;
+    --color-blue-7: #00489f;
+    --color-green-1: opacity(var(--color-green-4), 20%);
+    --color-green-2: opacity(var(--color-green-4), 40%);
+    --color-green-3: opacity(var(--color-green-4), 60%);
```
Comment (ArtBlue) [inline src/tokens/ds4-core.css:41]: Same percentages comment as previous about browser support.

File: docs/static/ds4-core.css [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    --color-neutral-0: #fff;
+    --color-neutral-1: #eee;
+    --color-neutral-2: #ddd;
+    --color-neutral-3: #ccc;
+    --color-neutral-4: #999;
+    --color-neutral-5: #707070;
+    --color-neutral-6: #333;
+    --color-neutral-7: #111820;
+    --color-neutral-8: #000;
+    --color-orange-1: opacity(var(--color-orange-4), 20%);
```
Comment (ArtBlue) [inline docs/static/ds4-core.css:11]: Opacity percentage issue

File: src/less/mixins/utility-mixins.less [hunk 3]
```diff
@@ -63,11 +63,11 @@
 // token system
 
 .customPropertyToken(@property, @token) {
-    @{property}: var(~"--@{token}", @@token);
+    @{property}: var(~"--@{token}");
```
[the remark this answers is not in the document]
Comment (agliga) [reply src/less/mixins/utility-mixins.less:66]: Yes, I agree, we don't need this one, but we still need the fallback ones.

[the remark this answers is not in the document]
Comment (ianmcburnie) [reply src/less/mixins/utility-mixins.less:66]: The idea is that they will all be removed. Even the fallback ones. See notes in description

File: src/less/mixins/dialog-mixins.less [hunk 4]
```diff
@@ -3,8 +3,8 @@
 @dialog-border-radius: @border-radius-100;
 @dialog-small-gutter-size: 16px;
 @dialog-large-gutter-size: 24px;
-@dialog-scrim-color-hide: fade(@color-foreground-primary, 0%);
-@dialog-scrim-color-show: fade(@color-foreground-primary, 70%);
+@dialog-scrim-color-hide: opacity(var(--color-foreground-primary), 0%);
+@dialog-scrim-color-show: opacity(var(--color-foreground-primary), 70%);
```
[the remark this answers is not in the document]
Comment (ianmcburnie) [reply src/less/mixins/dialog-mixins.less:7]: Ah yes, I just quickly switched it over from the LESS fade function to pure CSS to get it working. Will change.

**Review by ArtBlue — APPROVED** [review]
The general direction is good. If we accomplish the full move to css props without compromising the final file sizes and performance too much, this would be ideal.

**Comment by ianmcburnie** [comment]
So the main change that I think we need consensus/go-ahead on is the fact that there are no "default" values anymore. If a custom property value isn't found, things will look broken in many places. This is intentional, but a big deviation from how we had it before with the LESS variables that acted as the defaults/fallbacks.

**Comment by ArtBlue** [comment]
> So the main change that I think we need consensus/go-ahead on is the fact that there are no "default" values anymore. If a custom property value isn't found, things will look broken in many places. This is intentional, but a big deviation from how we had it before with the LESS variables that acted as the defaults/fallbacks.

I'm in favor of that change. I think it's better to be explicit when setting values. That approach tends to promote more clarity and better organization, and leaves far less to be dealt with at the tail end of the implementation.

**Pull request merged**